src/core/timings.js

```diff
@@ -162,7 +162,7 @@ function timingsSampler(video) {
         playback: currentTimings.playback,
       };
     }
-    else if (wasStalled && currentGap < Infinity && currentGap > resumeGap(wasStalled)) {
+    else if (wasStalled && currentGap < Infinity && (currentGap > resumeGap(wasStalled) || ending)) {
```

Comment:
I also added this because if the player stops playing due to network issues 4 seconds before the end of the video, the player would never start playing again as we would never reach the resume gap.